### ISSUE TITLE
Typing "<" gets encoded as <S-lt>

### ISSUE THREAD

**Issue by equalsraf** [equalsraf/neovim-qt#134]
Reported by @khalidchawtany. In Mac OS X US layout `<` is inserted by typing `Shift`+`,` but instead of getting \<lt\> neovim receives a `<S-lt>`. This happens because in input.c we remove the SHIFT modifier later in the function i.e.
1. Removal of SHIFT happens after `<` is processed
2. `<` is handled twice, one time as part of the specialKeys and another as a special escape case, and this case needs to be handled before either
3. Presumably this could also happen in layouts where `\` is the result of pressing shift and another key

Its a bit strange that on a Mac Qt reports this event key: `<` with modifier `Shift` and text: `<`

Tentative patch

``` diff
diff --git a/src/gui/input.cpp b/src/gui/input.cpp
index b517c4b..97cb019 100644
--- a/src/gui/input.cpp
+++ b/src/gui/input.cpp
@@ -150,6 +150,10 @@ QString InputConv::convertMouse(Qt::MouseButton bt, QEvent::Type type, Qt::Keybo
  */
 QString InputConv::convertKey(const QString& text, int k, Qt::KeyboardModifiers mod)
 {
+       if (text == "<" || k == Qt::Key_Less) {
+               mod &= ~ShiftModifier;
+       }
+
        if (specialKeys.contains(k)) {
                return QString("<%1%2>").arg(modPrefix(mod)).arg(specialKeys.value(k));
        }
```

**Comment by lambdalisue** [comment]
I had same issue on #160.

Interestingly, my environment is a bit different from what mentioned above.

OS: Ubuntu 16.04 Gnome desktop 64bit
Neovim: NVIM v0.1.5-634-g0595da8
Keyboard: [109 JP layout](http://www2d.biglobe.ne.jp/~msyk/keyboard/layout/109key.png)

I tested several keys but only `<` does not work correctly.

```
"/" 47 QFlags<Qt::KeyboardModifiers>(NoModifier) "/"
"/" 47 QFlags<Qt::KeyboardModifiers>(NoModifier) "/"
"/" 47 QFlags<Qt::KeyboardModifiers>(NoModifier) "/"
"\u001B" 16777216 QFlags<Qt::KeyboardModifiers>(NoModifier) "<Esc>"
"?" 63 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "?"
"?" 63 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "?"
"?" 63 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "?"
"\\" 92 QFlags<Qt::KeyboardModifiers>(NoModifier) "<Bslash>"
"\\" 92 QFlags<Qt::KeyboardModifiers>(NoModifier) "<Bslash>"
"\\" 92 QFlags<Qt::KeyboardModifiers>(NoModifier) "<Bslash>"
"_" 95 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "_"
"_" 95 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "_"
"_" 95 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "_"
"<" 60 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "<S-lt>"
"<" 60 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "<S-lt>"
"<" 60 QFlags<Qt::KeyboardModifiers>(ShiftModifier) "<S-lt>"
"," 44 QFlags<Qt::KeyboardModifiers>(NoModifier) ","
"," 44 QFlags<Qt::KeyboardModifiers>(NoModifier) ","
"," 44 QFlags<Qt::KeyboardModifiers>(NoModifier) ","
">" 62 QFlags<Qt::KeyboardModifiers>(ShiftModifier) ">"
">" 62 QFlags<Qt::KeyboardModifiers>(ShiftModifier) ">"
">" 62 QFlags<Qt::KeyboardModifiers>(ShiftModifier) ">"
```

**Comment by eyalk5** [comment]
I have quite a new version of nvim-qt, and it happens(two month ago probably).  

It prevents me from typing < on leaderF.

(I don't know how to check the version BTW).

**Comment by lambdalisue** [comment]
#162 fixed issue at that time. The changes is small so migrate it by yourself to make a new PR. I'm OK that someone take my PR over.

**Comment by eyalk5** [comment]
Well, this patch is already incorporated into the master of neovim-qt. Just not directly.
The issue persists. And I couldn't find a way to encode "<" in a way that it wouldn't interpret this as the opening clause of a special command. This issue need to be dealt with in another place.

**Comment by jgehrig** [comment]
This issue is resolved in the Pull Request above. It is available in `v0.2.16`

Marking as closed.
